apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.js

```diff
@@ -1,20 +1,25 @@
 // @ts-check
-const express = require('express');
 const asyncHandler = require('express-async-handler');
-const path = require('path');
+import * as express from 'express';
+import * as path from 'path';
 
-const { flash } = require('@prairielearn/flash');
-const sqldb = require('@prairielearn/postgres');
-const { logger } = require('@prairielearn/logger');
-const Sentry = require('@prairielearn/sentry');
+import { flash } from '@prairielearn/flash';
+import * as sqldb from '@prairielearn/postgres';
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
 
-const opsbot = require('../../lib/opsbot');
-const github = require('../../lib/github');
-const { config } = require('../../lib/config');
+import * as opsbot from '../../lib/opsbot';
+import * as github from '../../lib/github';
+import { config } from '../../lib/config';
 
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
+router.use((req, res, next) => {
+  res.locals.navPage = 'request_course';
+  next();
+});
```

Comment:
There's no need for separate middleware here; there's only a single `GET` handler here, so this can be assigned directly in that handler. This is what the home page does:

https://github.com/PrairieLearn/PrairieLearn/blob/1b07ae881b66fbfff3b57ac1f824e3d0a671214c/apps/prairielearn/src/pages/home/home.js#L11

Reply:
I did that but then put it back in the middleware. POST might use it too on an error. I'd like to move this page to `.html.ts` soon enough anyway so I'm not worried about just forklifting the middleware over, but we could put it in both `router` calls too.

Reply:
We'll never respond to a `POST` with anything but a redirect, so we don't ever need to set `res.locals.navPage` for a `POST` handler.